integration/test_ml.py

```diff
@@ -0,0 +1,361 @@
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Integration tests for firebase_admin.ml module."""
+import re
+import os
+import shutil
+import random
+import tempfile
+import pytest
+
+
+from firebase_admin import ml
+from firebase_admin import exceptions
+from tests import testutils
+
+
+# pylint: disable=import-error,no-name-in-module
+try:
+    import tensorflow as tf
+    _TF_ENABLED = True
+except ImportError:
+    _TF_ENABLED = False
+
+
+NAME_ONLY_ARGS = {
+    'display_name': 'TestModel123_{0}'.format(random.randint(1111, 9999))
+}
+NAME_ONLY_ARGS_UPDATED = {
```

Comment:
Now that we have done the refactor, I wonder if it makes sense to define each of these as a fixture.

```
@pytest.fixture
def name_only_model():
  args = {
    'display_name': _random_identifier('TestModel')
  }
  model = ml.create_model(ml.Model(**args))
  yield model, args
  _clean_up_model(model)

@pytest.fixture(scope='module')
def tflite_format():
  # Create the model format from the file and return
  # Thought defining this bit as a module-scoped fixture will speed things up a bit

@pytest.fixture
def full_model(tflite_format): # Here we reference the tflite_format fixture
  args = {
    'display_name': _random_identifier('TestFullModel'),
    'tags': ['test1', 'test'2],
    'model_format': tflite_format,
  }
  model = ml.create_model(ml.Model(**args))
  yield model, args
  _clean_up_model(model)

def test_create_simple_model(name_only_model):
  model, args = name_only_model
  check_model(model, args) 
```

That simplifies the call-site significantly, and makes sure multiple calls to the same fixture don't reuse identifiers.

Reply:
We started with something very similar to this. I think we've done enough refactoring now.